cunumeric/eager.py

```diff
@@ -820,7 +820,12 @@ def cholesky(self, src, no_tril):
         if self.deferred is not None:
             self.deferred.cholesky(src, no_tril)
         else:
-            result = np.linalg.cholesky(src.array)
+            try:
+                result = np.linalg.cholesky(src.array)
+            except np.linalg.LinAlgError as e:
+                from .linalg import LinAlgError
+
+                raise LinAlgError(e)
```

Comment:
I suggest we change this to `raise LinAlgError(e) from e`. Then instead of seeing:

```
numpy.linalg.LinAlgError: Matrix is not positive definite

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
...
cunumeric.linalg.exception.LinAlgError: Matrix is not positive definite
```

we see:

```
numpy.linalg.LinAlgError: Matrix is not positive definite

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
...
cunumeric.linalg.exception.LinAlgError: Matrix is not positive definite
```

Reply:
682656c9027dac1934ad212fd2bb5ea31648efa3